WordPress/src/main/java/org/wordpress/android/ui/posts/prepublishing/visibility/usecases/UpdatePostStatusUseCase.kt

```diff
@@ -31,6 +32,12 @@ class UpdatePostStatusUseCase @Inject constructor() {
 
         editPostRepository.updateAsync({ postModel: PostModel ->
             postModel.setStatus(postStatus.toString())
+
+            // since PRIVATE posts always have to be immediate we set the current date when updating to be sure.
```

Comment:
Do you please have a link to the location where this logic was located before?

>to be sure

Imho it's good practice to avoid adding code "to be sure". It either needs to be there or it doesn't. Including these types of "safety" branches makes the code very difficult to refactor in the future. It basically says that even the original author of the code wasn't sure if it was necessary - it's ten times more difficult to find out for the person who wants to re-use/refactor the code. Wdyt?

Reply:
> Do you please have a link to the location where this logic was located before?

So I came up with this logic on my own based on the discussion we had in the prepublishing nudges channel. This is a new approach we are utilizing to ensure that the `PRIVATE` posts can't be scheduled or to be more specific, `PRIVATE` posts won't give the impression that they are scheduled. 

> Imho it's good practice to avoid adding code "to be sure". It either needs to be there or it doesn't. Including these types of "safety" branches makes the code very difficult to refactor in the future. It basically says that even the original author of the code wasn't sure if it was necessary - it's ten times more difficult to find out for the person who wants to re-use/refactor the code. Wdyt?

This is a really good point. So answer your main concern this code definitely needs to be there. When I said just to be sure I meant that a post can be in the state where it's being published `Immediately` but it can also be in a state where it was being `SCHEDULED` and then the `PostStatus` is changed to `Private` so we should always clear a future date if a post is set to a `PostStatus` of `PRIVATE`. This logic will also need to be shared as well, so I am thinking of moving it somewhere else because the `PostSettings` should have this behavior.

Reply:
Thanks for the clarification! ;)

>This logic will also need to be shared as well, so I am thinking of moving it somewhere else because the PostSettings should have this behavior.

That'd be great! ;) 

Just a double-check, I don't recall how these creation dates work. Won't this code modify the date even when we edit a year old private post or is it a desired behavior?

Reply:
> That'd be great! ;)
Awesome man 😄 

>Just a double-check, I don't recall how these creation dates work. Won't this code modify the date even when we edit a year old private post or is it a desired behavior?

This a really good point. We should check in with the iOS @leandroalonso  about this as well. I just looked at their PR and it doesn't seem like they took this into consideration. I am trying to think of another approach that we could take. How about only setting it to `Immediately` if the date was `Scheduled` i.e the date is scheduled for the future? So if the user selects a date that's in the past or modifies a post in the past that date is still shown?

Reply:
> Won't this code modify the date even when we edit a year old private post or is it a desired behavior?

At least in iOS, yes, this will change the date. And I'd say no, it's not a desired behavior. 😬

Reply:
Awesome. Love how we addressed this team @leandroalonso @malinajirka 😄